compatibility-verifier/compCheck.sh

```diff
@@ -0,0 +1,210 @@
+    echo "Usage: $command serviceName"
+  fi
+  exit 1
+}
+
+# cleanup the temporary directory when exiting the script
+function cleanup() {
+  if [ -n "$tmpDir" ] && [ -d "$tmpDir" ]; then
+    rm -rf "$tmpDir"
+  fi
+}
+
+# cleanp the temporary directory when the bash script exits 
+trap cleanup EXIT
+
+if [ $# -lt 2 ] || [ $# -gt 3 ] ; then
+  usage compCheck
+fi
+
+# get arguments
+olderCommit=$1
+newerCommit=$2 
+
+if [ -n "$3" ]; then
+  workingDir=$3
+  if [ -d "$workingDir" ]; then
+    echo "Directory ${workingDir} already exists. Use a new directory."
+    exit 1
+  fi
+else
+  # use a temp directory in case workingDir is not provided
+  tmpDir=$(mktemp -d 2>/dev/null || mktemp -d -t 'mytmpdir')
+  workingDir=$tmpDir
+fi
+
+
+# This function builds Pinot given a specific commit hash and target directory
+function checkoutAndBuild() {
+  if [ $# -ne 2 ]; then 
+    usage checkoutAndBuild
+  fi
+  commitHash=$1
+  targetDir=$2
+  
+  pushd "$targetDir" || exit 1
+  git init
+  git remote add origin https://github.com/apache/incubator-pinot
+  git fetch --depth 1 origin "$commitHash"
+  git checkout FETCH_HEAD
+  mvn install package -DskipTests -Pbin-dist
+  popd || exit 1
+}
+
+# Given a component and directory, start that version of the specific component 
+function startService() {
+  if [ $# -ne 2 ]; then 
+    usage startService
+  fi
+  serviceName=$1
+  dirName=$2
+  if [[ ! " ${components[*]} " =~ $serviceName ]]; then
+    printf "Not a valid component. Needs to be one of: %s \n" "${components[*]}"
+    exit 1
+  fi
+  # Upon start, save the pid of the process for a component into a file in /tmp/{component}.pid, which is then used to stop it
+  pushd "$dirName"/pinot-tools/target/pinot-tools-pkg/bin  || exit 1
+  if [ "$serviceName" = "zookeeper" ]; then
+    sh -c 'echo $$ > /tmp/zookeeper.pid; exec ./pinot-admin.sh StartZookeeper' &
+  elif [ "$serviceName" = "controller" ]; then 
+    sh -c 'echo $$ > /tmp/controller.pid; exec ./pinot-admin.sh StartController' &  
+  elif [ "$serviceName" = "broker" ]; then
+    sh -c 'echo $$ > /tmp/broker.pid; exec ./pinot-admin.sh StartBroker' &
+  elif [ "$serviceName" = "server" ]; then
+    sh -c 'echo $$ > /tmp/server.pid; exec ./pinot-admin.sh StartServer' &
+  fi 
+  popd || exit 1
+}
+
+# Given a component, check if it known to be running and stop that specific component
+function stopService() {
+  if [ $# -ne 1 ]; then 
+    usage stopService
+  fi
+  serviceName=$1
+  if [[ ! " ${components[*]} " =~ $serviceName ]]; then
+    printf "Not a valid component. Needs to be one of: %s \n" "${components[*]}"
+    exit 1
+  fi
+  if [ -f /tmp/"${serviceName}".pid ]; then 
+    servicePid=$(</tmp/"${serviceName}".pid)
+    rm /tmp/"${serviceName}".pid
+    if [ -n "$servicePid" ]; then
+      kill -9 "$servicePid"
+    fi
+  else
+    echo "Can't stop component ${serviceName} since can't component ${serviceName} is not known to have been started."
```

Comment:
add the full path name to pid file in this message. "Pid file {} not found. Failed to stop component {}"